ocfweb/docs/docs/staff/backend/kubernetes.md

```diff
@@ -0,0 +1,190 @@
+[[!meta title="Kubernetes"]]
+
+At the OCF we are in the process of moving production services from
+[[Mesos/Marathon|doc staff/backend/mesos]] to [Kubernetes][kubernetes]. In this
+document we will explain the design of our Kubernetes cluster while also
+touching briefly on relevant core concepts. This page is _not_ a `HOWTO` for
+deploying services or troubleshooting a bad cluster. Rather, it is meant to
+explain architectural considerations such that current work can be built upon.
+Although, reading this document will help you both deploy services in the OCF
+Kubernetes cluster and debug issues when they arise.
+
+## Kubernetes
+
+Kubernetes is a container orchestration system open sourced by Google. Its main
+purpose is to schedule services to run on a cluster of computers while
+abstracting away the existence of the cluster from the services. The design of
+Kubernetes is loosely based on Google's internal orchestration system Borg.
+Kubernetes is now maintained by the [Cloud Native Computing Foundation][cncf],
+which is a part of the Linux Foundation. Kubernetes can flexibly handle
+replication, impose resource limits, and recover quickly from failures.
+
+## Kubernetes Cluster
+
+A Kubernetes cluster consists of "master" nodes and "worker" nodes. In short,
+master nodes share state to manage the cluster and schedule jobs to run on
+workers. It is considered best practice to run an odd number of masters, and
+currently our cluster has three masters.
+
+### Masters
+
+Kubernetes masters share state via a distributed key-value store. The reason
+for the key-value store is high availability: if one master goes down, a new
+master node will subsume control of the cluster in a leader election to ensure
+that no services incur downtime. [etcd][etcd-io] is an
```

Comment:
I'd write this something like the following:

> Kubernetes masters share state via etcd, a distributed key-value store implementing the Raft protocol. One master is elected as a leader for the cluster, with the ability to commit writes to the KVS. etcd then reliably replicates this state across every master, so that if the leader fails, another master can be elected and no state will be lost in the process. The state stored in etcd is scheduling state, service locations, and metadata, not state of the services running on the cluster itself.